DNN Platform/Modules/DDRMenu/MenuBase.cs

```diff
@@ -197,22 +197,15 @@ private void FilterNodes(string nodeString, bool exclude)
 			    }
 				else
 				{
-					var nodeText2 = nodeText;
-					filteredNodes.AddRange(
-						RootNode.Children.FindAll(
-							n =>
-							{
-								var nodeName = n.Text.ToLowerInvariant();
-								var nodeId = n.TabId.ToString();
-								return (nodeText2 == nodeName || nodeText2 == nodeId);
-							}));
+					var nodeText2 = nodeText;                    
+                    filteredNodes.Add(RootNode.FindByNameOrId(nodeText2));
 				}
 			}
 
             // if filtered for foksonomy tags, use flat tree to get all related pages in nodeselection
 		    if (flattenedNodes.HasChildren())
 		        RootNode = flattenedNodes;
-			RootNode.Children.RemoveAll(n => filteredNodes.Contains(n) == exclude);

```

Comment:
I see that the `exclude` parameter is not longer used in this method but it is called with both `true` and `false` in `MenuBase#PreRender`. If you're certain that your changes are correct for all cases please refactor the calling code too.

Reply:
Good catch that was surely missed out, thnx!